src/DeviceInterfaceValues.tsx

```diff
@@ -554,6 +645,26 @@ export default (): React.ReactElement => {
       });
   };
 
+  const getInterfaceData = (from: string, to: string) => {
```

Comment:
we already have a `fetcher` that takes care of the data, we could edit that one instead of implementing a new one.
Or is there something that makes it difficult to use that function?

Reply:
Are you referring to the ```deviceDataFetcher``` function? If yes, how to provide there ```since``` and ```to``` properties..

Reply:
the fetcher is defined before with
```javascript
  const deviceDataFetcher = useFetch(() =>
    astarte.client.getDeviceDataTree({
      deviceId,
      interfaceName,
    }),
  );
```
`useFetch` is one of our hooks that abstracts the logic of loading state (loading, error, ok). Have a look at `useFetch.tsx`.
The limitation is that the params cannot change after the hook is initialized.

Here i see 2 options:
- edit the hook so that it allows dynamic params (complex with little benefits)
- remove this hook entirely and rely directly on the Astarte client

I personally would go for the second one, which is also partially what you did in your commit, but i would have only a single `fetch` function with optional parameters, instead of multiple fetch functions